flang/module/iso_c_binding.f90

```diff
@@ -22,6 +22,12 @@ module iso_c_binding
     c_sizeof => sizeof, &
     operator(==), operator(/=)
 
+  implicit none
+
```

Comment:
It might be better to add the `private` statement on line 26 so that everything is private and then add explicit `public` statements for the things you want to be public.  This way, future additions don't leak by default.

Reply:
Good point! Even though it's way more to change, I like the suggestion as it's future proof.  Please have a look at the commit.